api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/util/RequestLoggingFilter.java

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+package org.hawkular.metrics.api.jaxrs.util;
+
+import static org.hawkular.metrics.api.jaxrs.config.ConfigurationKey.REQUEST_LOGGING_LEVEL;
+import static org.hawkular.metrics.api.jaxrs.config.ConfigurationKey.REQUEST_LOGGING_LEVEL_WRITES;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.PathSegment;
+import javax.ws.rs.ext.Provider;
+
+import org.hawkular.metrics.api.jaxrs.config.Configurable;
+import org.hawkular.metrics.api.jaxrs.config.ConfigurationProperty;
+import org.jboss.logging.Logger;
+
+/**
+ * @author jsanda
+ */
+@Logged
+@Provider
+public class RequestLoggingFilter implements ContainerRequestFilter {
+
+    private static Logger logger = Logger.getLogger(RequestLoggingFilter.class);
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(REQUEST_LOGGING_LEVEL)
+    private String loggingLevelConfig;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(REQUEST_LOGGING_LEVEL_WRITES)
+    private String writesLoggingLevelConfig;
+
+    private Logger.Level logLevel;
+
+    private Logger.Level writesLogLevel;
+
+    @PostConstruct
+    public void init() {
+        logLevel = getLogLevel(loggingLevelConfig);
+        writesLogLevel = getLogLevel(writesLoggingLevelConfig);
+    }
+
+    private Logger.Level getLogLevel(String config) {
+        if (config == null) {
+            return  null;
+        }
+        switch (config) {
+            case "TRACE": return Logger.Level.TRACE;
+            case "INFO": return Logger.Level.INFO;
+            case "WARN": return Logger.Level.WARN;
+            case "ERROR": return Logger.Level.ERROR;
+            case "FATAL": return Logger.Level.FATAL;
+            default: return Logger.Level.DEBUG;
+        }
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        try {
+            if (isInsertDataRequest(requestContext)) {
+                if (writesLogLevel != null) {
+                    logRequest(writesLogLevel, requestContext);
+                }
+            } else if (logLevel != null) {
+                logRequest(logLevel, requestContext);
+            }
+        } catch (Exception e) {
+            logger.info("Failed to log request", e);
+        }
+    }
+
+    private boolean isInsertDataRequest(ContainerRequestContext requestContext) {
+        List<PathSegment> pathSegments = requestContext.getUriInfo().getPathSegments();
+        return (requestContext.getMethod().equals("POST")) && !pathSegments.get(pathSegments.size() - 1).getPath()
+                .equals("query");
+    }
+
+    private void logRequest(Logger.Level level, ContainerRequestContext requestContext) {
+        String request =
```

Comment:
Because this is going to be a large string it is better to use something more efficient for constructing the string.